File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3StopStaffing.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3StopStaffing {
    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="fri_am_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "fri_am_from")]
    public string FriAmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="fri_am_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "fri_am_to")]
    public string FriAmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="fri_pm_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "fri_pm_from")]
    public string FriPmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="fri_pm_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "fri_pm_to")]
    public string FriPmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="mon_am_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "mon_am_from")]
    public string MonAmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="mon_am_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "mon_am_to")]
    public string MonAmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="mon_pm_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "mon_pm_from")]
    public string MonPmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="mon_pm_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "mon_pm_to")]
    public string MonPmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="ph_additional_text", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ph_additional_text")]
    public string PhAdditionalText { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="ph_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ph_from")]
    public string PhFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="ph_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "ph_to")]
    public string PhTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sat_am_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sat_am_from")]
    public string SatAmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sat_am_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sat_am_to")]
    public string SatAmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sat_pm_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sat_pm_from")]
    public string SatPmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sat_pm_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sat_pm_to")]
    public string SatPmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sun_am_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sun_am_from")]
    public string SunAmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sun_am_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sun_am_to")]
    public string SunAmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sun_pm_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sun_pm_from")]
    public string SunPmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="sun_pm_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "sun_pm_to")]
    public string SunPmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="thu_am_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "thu_am_from")]
    public string ThuAmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="thu_am_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "thu_am_to")]
    public string ThuAmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="thu_pm_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "thu_pm_from")]
    public string ThuPmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="thu_pm_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "thu_pm_to")]
    public string ThuPmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="tue_am_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tue_am_from")]
    public string TueAmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="tue_am_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tue_am_to")]
    public string TueAmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="tue_pm_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tue_pm_from")]
    public string TuePmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="tue_pm_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tue_pm_to")]
    public string TuePmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="wed_am_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "wed_am_from")]
    public string WedAmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="wed_am_to", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "wed_am_to")]
    public string WedAmTo { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="wed_pm_from", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "wed_pm_from")]
    public string WedPmFrom { get; set; }

    /// <summary>
    /// Stop staffing hours
    /// </summary>
    /// <value>Stop staffing hours</value>
    [DataMember(Name="wed_pm_To", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "wed_pm_To")]
    public string WedPmTo { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3StopStaffing {\n");
      sb.Append("  FriAmFrom: ").Append(FriAmFrom).Append("\n");
      sb.Append("  FriAmTo: ").Append(FriAmTo).Append("\n");
      sb.Append("  FriPmFrom: ").Append(FriPmFrom).Append("\n");
      sb.Append("  FriPmTo: ").Append(FriPmTo).Append("\n");
      sb.Append("  MonAmFrom: ").Append(MonAmFrom).Append("\n");
      sb.Append("  MonAmTo: ").Append(MonAmTo).Append("\n");
      sb.Append("  MonPmFrom: ").Append(MonPmFrom).Append("\n");
      sb.Append("  MonPmTo: ").Append(MonPmTo).Append("\n");
      sb.Append("  PhAdditionalText: ").Append(PhAdditionalText).Append("\n");
      sb.Append("  PhFrom: ").Append(PhFrom).Append("\n");
      sb.Append("  PhTo: ").Append(PhTo).Append("\n");
      sb.Append("  SatAmFrom: ").Append(SatAmFrom).Append("\n");
      sb.Append("  SatAmTo: ").Append(SatAmTo).Append("\n");
      sb.Append("  SatPmFrom: ").Append(SatPmFrom).Append("\n");
      sb.Append("  SatPmTo: ").Append(SatPmTo).Append("\n");
      sb.Append("  SunAmFrom: ").Append(SunAmFrom).Append("\n");
      sb.Append("  SunAmTo: ").Append(SunAmTo).Append("\n");
      sb.Append("  SunPmFrom: ").Append(SunPmFrom).Append("\n");
      sb.Append("  SunPmTo: ").Append(SunPmTo).Append("\n");
      sb.Append("  ThuAmFrom: ").Append(ThuAmFrom).Append("\n");
      sb.Append("  ThuAmTo: ").Append(ThuAmTo).Append("\n");
      sb.Append("  ThuPmFrom: ").Append(ThuPmFrom).Append("\n");
      sb.Append("  ThuPmTo: ").Append(ThuPmTo).Append("\n");
      sb.Append("  TueAmFrom: ").Append(TueAmFrom).Append("\n");
      sb.Append("  TueAmTo: ").Append(TueAmTo).Append("\n");
      sb.Append("  TuePmFrom: ").Append(TuePmFrom).Append("\n");
      sb.Append("  TuePmTo: ").Append(TuePmTo).Append("\n");
      sb.Append("  WedAmFrom: ").Append(WedAmFrom).Append("\n");
      sb.Append("  WedAmTo: ").Append(WedAmTo).Append("\n");
      sb.Append("  WedPmFrom: ").Append(WedPmFrom).Append("\n");
      sb.Append("  WedPmTo: ").Append(WedPmTo).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/lib/model/v3_stop_staffing.dart
part of swagger.api;

class V3StopStaffing {
  /* Stop staffing hours */
  String friAmFrom = null;
/* Stop staffing hours */
  String friAmTo = null;
/* Stop staffing hours */
  String friPmFrom = null;
/* Stop staffing hours */
  String friPmTo = null;
/* Stop staffing hours */
  String monAmFrom = null;
/* Stop staffing hours */
  String monAmTo = null;
/* Stop staffing hours */
  String monPmFrom = null;
/* Stop staffing hours */
  String monPmTo = null;
/* Stop staffing hours */
  String phAdditionalText = null;
/* Stop staffing hours */
  String phFrom = null;
/* Stop staffing hours */
  String phTo = null;
/* Stop staffing hours */
  String satAmFrom = null;
/* Stop staffing hours */
  String satAmTo = null;
/* Stop staffing hours */
  String satPmFrom = null;
/* Stop staffing hours */
  String satPmTo = null;
/* Stop staffing hours */
  String sunAmFrom = null;
/* Stop staffing hours */
  String sunAmTo = null;
/* Stop staffing hours */
  String sunPmFrom = null;
/* Stop staffing hours */
  String sunPmTo = null;
/* Stop staffing hours */
  String thuAmFrom = null;
/* Stop staffing hours */
  String thuAmTo = null;
/* Stop staffing hours */
  String thuPmFrom = null;
/* Stop staffing hours */
  String thuPmTo = null;
/* Stop staffing hours */
  String tueAmFrom = null;
/* Stop staffing hours */
  String tueAmTo = null;
/* Stop staffing hours */
  String tuePmFrom = null;
/* Stop staffing hours */
  String tuePmTo = null;
/* Stop staffing hours */
  String wedAmFrom = null;
/* Stop staffing hours */
  String wedAmTo = null;
/* Stop staffing hours */
  String wedPmFrom = null;
/* Stop staffing hours */
  String wedPmTo = null;

  V3StopStaffing();

  @override
  String toString() {
    return 'V3StopStaffing[friAmFrom=$friAmFrom, friAmTo=$friAmTo, friPmFrom=$friPmFrom, friPmTo=$friPmTo, monAmFrom=$monAmFrom, monAmTo=$monAmTo, monPmFrom=$monPmFrom, monPmTo=$monPmTo, phAdditionalText=$phAdditionalText, phFrom=$phFrom, phTo=$phTo, satAmFrom=$satAmFrom, satAmTo=$satAmTo, satPmFrom=$satPmFrom, satPmTo=$satPmTo, sunAmFrom=$sunAmFrom, sunAmTo=$sunAmTo, sunPmFrom=$sunPmFrom, sunPmTo=$sunPmTo, thuAmFrom=$thuAmFrom, thuAmTo=$thuAmTo, thuPmFrom=$thuPmFrom, thuPmTo=$thuPmTo, tueAmFrom=$tueAmFrom, tueAmTo=$tueAmTo, tuePmFrom=$tuePmFrom, tuePmTo=$tuePmTo, wedAmFrom=$wedAmFrom, wedAmTo=$wedAmTo, wedPmFrom=$wedPmFrom, wedPmTo=$wedPmTo, ]';
  }

  V3StopStaffing.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    friAmFrom = json['fri_am_from'];
    friAmTo = json['fri_am_to'];
    friPmFrom = json['fri_pm_from'];
    friPmTo = json['fri_pm_to'];
    monAmFrom = json['mon_am_from'];
    monAmTo = json['mon_am_to'];
    monPmFrom = json['mon_pm_from'];
    monPmTo = json['mon_pm_to'];
    phAdditionalText = json['ph_additional_text'];
    phFrom = json['ph_from'];
    phTo = json['ph_to'];
    satAmFrom = json['sat_am_from'];
    satAmTo = json['sat_am_to'];
    satPmFrom = json['sat_pm_from'];
    satPmTo = json['sat_pm_to'];
    sunAmFrom = json['sun_am_from'];
    sunAmTo = json['sun_am_to'];
    sunPmFrom = json['sun_pm_from'];
    sunPmTo = json['sun_pm_to'];
    thuAmFrom = json['thu_am_from'];
    thuAmTo = json['thu_am_to'];
    thuPmFrom = json['thu_pm_from'];
    thuPmTo = json['thu_pm_to'];
    tueAmFrom = json['tue_am_from'];
    tueAmTo = json['tue_am_to'];
    tuePmFrom = json['tue_pm_from'];
    tuePmTo = json['tue_pm_to'];
    wedAmFrom = json['wed_am_from'];
    wedAmTo = json['wed_am_to'];
    wedPmFrom = json['wed_pm_from'];
    wedPmTo = json['wed_pm_To'];
  }

  Map<String, dynamic> toJson() {
    return {
      'fri_am_from': friAmFrom,
      'fri_am_to': friAmTo,
      'fri_pm_from': friPmFrom,
      'fri_pm_to': friPmTo,
      'mon_am_from': monAmFrom,
      'mon_am_to': monAmTo,
      'mon_pm_from': monPmFrom,
      'mon_pm_to': monPmTo,
      'ph_additional_text': phAdditionalText,
      'ph_from': phFrom,
      'ph_to': phTo,
      'sat_am_from': satAmFrom,
      'sat_am_to': satAmTo,
      'sat_pm_from': satPmFrom,
      'sat_pm_to': satPmTo,
      'sun_am_from': sunAmFrom,
      'sun_am_to': sunAmTo,
      'sun_pm_from': sunPmFrom,
      'sun_pm_to': sunPmTo,
      'thu_am_from': thuAmFrom,
      'thu_am_to': thuAmTo,
      'thu_pm_from': thuPmFrom,
      'thu_pm_to': thuPmTo,
      'tue_am_from': tueAmFrom,
      'tue_am_to': tueAmTo,
      'tue_pm_from': tuePmFrom,
      'tue_pm_to': tuePmTo,
      'wed_am_from': wedAmFrom,
      'wed_am_to': wedAmTo,
      'wed_pm_from': wedPmFrom,
      'wed_pm_To': wedPmTo
     };
  }

  static List<V3StopStaffing> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3StopStaffing>() : json.map((value) => new V3StopStaffing.fromJson(value)).toList();
  }

  static Map<String, V3StopStaffing> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3StopStaffing>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3StopStaffing.fromJson(value));
    }
    return map;
  }
}
